src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -231,6 +233,12 @@ internal static class PeWriter
             var strongNameProvider = context.Module.CommonCompilation.Options.StrongNameProvider;
             var corFlags = properties.CorFlags;
 
+            if (managedResourceBuilder.Count == 0)
```

Comment:
There are essentially two cases here:

1. The `managedResourceBuilder` has content in which case it will be linked, via `LinkSuffix`, to `peBlob` below. Thus freeing `peBlob` frees both. 
2. The `managedResourceBuilder` has no content thus `LinkSuffix` does nothing and is discarded. This means it is not returned to the pool. 

After some thought decided the easiest approach here was to just discard `managedResourceBuilder` here if it's unused. It's optional below so this is not a functional change. Happy to consider other approaches.

Reply:
Went ahead and filed a bug https://github.com/dotnet/runtime/issues/99266